common/ac.go

```
@@ -14,41 +15,36 @@ var (
 )
 
 type KeyConvert struct {
+	Origin string `json:"origin"`
 	Prefix string `json:"prefix"`
 }
 
 //Convert the format of args according to the cmd
 //1.should convert all keys used by the commands which may operate multiple keys simultaneously(mget eg.)
 //2.may use a pool of byte slice to reduce the cost of GC
 func (conv *KeyConvert) Convert(cmd string, args [][]byte) [][]byte {
-	if len(conv.Prefix) == 0 {
+	if len(conv.Origin) == 0 && len(conv.Prefix) == 0 {
 		return args
 	}
 
-	newArgs := make([][]byte, len(args))
-	copy(newArgs, args)
+	newArgs, err := ConvertRedisKey(cmd, args, func(key []byte) ([]byte, error) {
+		if len(conv.Origin) > 0 {
+			newKey := make([]byte, len(key))
```

Please use `string.HasPrefix` to check whether it is starting with the replacing prefix. If not, it should return an error message. And maybe optimize with below:
```
newKey := make([]byte, len(conv.Prefix)+len(key)-len(conv.Origin))
copy(newKey, conv.Prefix)
copy(newKey[len(conv.Prefix):], key[len(conv.Origin):])
return newKey, nil
```